source/common/network/win32_socket_handle_impl.cc

```diff
@@ -84,5 +125,67 @@ void Win32SocketHandleImpl::reEnableEventBasedOnIOResult(const Api::IoCallUint64
   }
 }
 
+Api::IoCallUint64Result Win32SocketHandleImpl::drainToPeekBuffer(size_t length) {
+  size_t total_bytes_read = 0;
+  while (peek_buffer_.length() < length) {
+    Buffer::Reservation reservation = peek_buffer_.reserveForRead();
+    uint64_t bytes_to_read =
+        std::min<uint64_t>(static_cast<uint64_t>(length - total_bytes_read), reservation.length());
```

Comment:
If we've previously peeked 10 bytes, and we want a total of 15, won't this read a full 15, rather than the difference of 5, or am I misreading?  If so please fix and test

Reply:
Since the previous 10 bytes already drain into the `peek_buffer_`, so we drain another 5 bytes. But in the end, we return total 15 bytes to the caller https://github.com/envoyproxy/envoy/pull/19291/files#diff-9a58b923da541e9bbe65ae41bf9d5a42cd1f5a123603fedd08fdade3fd658ac6R118

Reply:
emm... I think you are right, let me check and add test